src/orch/opt.c

```diff
@@ -5,25 +5,36 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-void get_opts(int argc, char *argv[], int *ctrlPort) {
+int get_opts(int argc, char *argv[], int *ctrlPort, char *configPath) {
         int opt;
-        int pflag;
-        while ((opt = getopt(argc, argv, "p:")) != -1) {
+        int pflag = 0;
+
+        while ((opt = getopt(argc, argv, "p:c:")) != -1) {
                 switch (opt) {
                 case 'p':
                         if (parse_long(optarg, (long *) ctrlPort) != 0) {
                                 fprintf(stderr, "Invalid port option '%s'\n", optarg);
-                                exit(EXIT_FAILURE);
+                                return EXIT_FAILURE;
                         }
                         pflag = 1;
                         break;
+                case 'c':
+                        strncpy(configPath, optarg, 1024);
+                        break;
                 default:
-                        exit(EXIT_FAILURE);
+                        return EXIT_FAILURE;
                 }
         }
 
         if (pflag != 1) {
                 fprintf(stderr, "Missing port option. Usage: %s [-p port]\n", argv[0]);
-                exit(EXIT_FAILURE);
+                return EXIT_FAILURE;
         }
+
+        if (configPath == NULL) {
+                fprintf(stderr, "Missing INI file path. Usage: %s [-c /path/to/ini-file]\n", argv[0]);
+                return EXIT_FAILURE;
+        }
+
+        return EXIT_SUCCESS;
```

Comment:
How about using `#include <stdbool.h>`, which defines `true` and `false`? This is just another syntactic sugar for 0 and 1, but feels more natural and fits 
this case better, I think. What do you think?

Reply:
That looks more clean and "natural" for me. I think it would be nice to decide this and describe it in our code style

Reply:
The "EXIT" functions are used when exiting the program, not for returning from functions.

Reply:
A suggestion: We're introducing a new function get_opts().  That function doesn't need to do much, and meanwhile, by having a new function, now we really should do things like check its arguments for NULL pointers.  I would suggest not breaking this out into a separate function, and instead put the getopt() in main().  The reason is, as a point of style, main() could be made to do only minimal stuff, such as check the command line arguments of the program, register signal handlers, go into an event loop, etc.  And this top-level stuff could be put in a separate file called main.c.  In any case, if we keep get_opts(), we need to add a NULL pointer check for ctrlPort, because it gets dereferenced in parse_long().